### ISSUE TITLE
Add support for keyboard navigation using Tab key to move focus to Popup when opened using ToggleButton

### ISSUE THREAD

**Issue by wieslawsoltes** [AvaloniaUI/Avalonia#5000]
When you open Popup using ToggleButton (pressing Space key, using Tab key for navigation) the focus is not moved inside popup. While you keep navigating with Tab key the popup stay open and does not receive keyboard focus.
```XAML
        <ToggleButton Name="PopupButton" Height="30">
            <Panel Height="30" DockPanel.Dock="Top">
                <Border Background="DarkGray" />
                </Border>
            </Panel>
        </ToggleButton>
        <Popup IsOpen="{Binding #PopupButton.IsChecked, Mode=TwoWay}" StaysOpen="False" Width="300">
            <Border>
                    <TextBox Text="Text" />
            </Border>
        </Popup>
```
